src/main/java/org/openstreetmap/atlas/utilities/caching/ResourceFetchFunction.java

```diff
@@ -0,0 +1,25 @@
+package org.openstreetmap.atlas.utilities.caching;
+
+import java.net.URI;
+
+import org.openstreetmap.atlas.streaming.resource.Resource;
+
+/**
+ * A {@link ResourceFetchFunction} is any function that takes a {@link URI} and returns the
+ * {@link Resource} at that {@link URI}. The function should return null if the resource cannot be
+ * retrieved.
+ *
+ * @author lcram
+ */
+@FunctionalInterface
+public interface ResourceFetchFunction
```

Comment:
Instead of having an interface for that, could this just be a `java.util.function.Function<URI, Resource>`?